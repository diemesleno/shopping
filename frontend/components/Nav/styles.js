import styled from 'styled-components';

export const Container = styled.ul`
  margin: 0;
  padding: 0;
  display: flex;
  justify-self: end;
  font-size: 2rem;
  text-decoration: none;
  list-style-type: none;
  li {
    text-decoration: none;
    text-transform: uppercase;
  }
  a,
  button {
    padding: 1rem 3rem;
    display: flex;
    align-items: center;
    position: relative;
    text-decoration: none;
    cursor: pointer;
    color: ${props => props.theme.black};
    @media (max-width: 700px) {
      font-size: 2rem;
      padding: 0 10px;
    }
    &:hover,
    &:focus {
      outline: none;
      &:after {
        width: calc(100% - 60px);
      }
    }
  }
  @media (max-width: 1300px) {
    border-top: 1px solid ${props => props.theme.lightgrey};
    width: 100%;
    justify-content: center;
    font-size: 1.5rem;
  }
`;
