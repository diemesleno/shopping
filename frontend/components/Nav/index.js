import Link from 'next/link';
import { Container } from './styles';
import User from '../User';
import Signout from '../Signout';

export default function Nav() {
  return (
    <User>
      {({ data }) => {
        const me = data ? data.me : null
        return (
          <Container>
            <li>
              <Link href="/items">
                <a>Shop</a>
              </Link>
            </li>
            {me && (
              <>
                <li>
                  <Link href="/sell">
                    <a>Sell</a>
                  </Link>
                </li>
                <li>
                  <Link href="/">
                    <a>Orders</a>
                  </Link>
                </li>
                <li>
                  <Signout />
                </li>
              </>
            )}
            {!me && (
              <li>
                <Link href="/signup">
                  <a>Sign In</a>
                </Link>
              </li>
            )}
          </Container>
        )
      }}
    </User>
  );
}