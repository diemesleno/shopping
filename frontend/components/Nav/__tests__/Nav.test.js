import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Nav from '../index';

describe('<Nav />', () => {
  it('renders', () => {
    const wrapper = shallow(<Nav />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
