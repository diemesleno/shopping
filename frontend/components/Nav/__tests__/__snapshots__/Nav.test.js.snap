// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders 1`] = `
<styles__Container>
  <li>
    <Link
      href="/items"
    >
      <a>
        Shop
      </a>
    </Link>
  </li>
  <li>
    <Link
      href="/"
    >
      <a>
        Sell
      </a>
    </Link>
  </li>
  <li>
    <Link
      href="/"
    >
      <a>
        Signup
      </a>
    </Link>
  </li>
  <li>
    <Link
      href="/"
    >
      <a>
        Orders
      </a>
    </Link>
  </li>
</styles__Container>
`;
